nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py

```diff
@@ -698,6 +706,8 @@ def __init__(
                     "Did not replace G2P valid symbol set since the given set is equivalent to the existing one."
                 )
                 self.set_fixed_vocab = False
+            elif read_phones_from_file:
+                tokens = get_ipa_character_set(locale)
             else:
                 g2p.replace_symbols(tokens)
```

Comment:
Fixed_vocab should only be set if the user wants to hardcode the tokens list. The check should instead act like:

```
if fixed_vocab:
    ...
elif read_phones_from_file:
    tokens = get_ipa_character_set(locale)
    self.set_fixed_vocab = False
else:
    ...
```